### PR TITLE
simplified nested finders

### PR DESCRIPTION
**Pull request by gharlan**
please take look. maybe I've overlooked something.

### CODE REVIEW

File: src/Console/ConfigurationResolver.php [hunk 1]
```diff
@@ -406,38 +406,26 @@ private function resolveConfigPath()
...
-                : Finder::create()->in($this->path)->getIterator()
-            ;
-
             $iterator = new \CallbackFilterIterator(
-                $nestedIterator,
-                function (\SplFileInfo $current) use ($currentIteratorItems) {
-                    return in_array($current->getRealPath(), $currentIteratorItems, true);
+                $currentFinder instanceof \IteratorAggregate ? $currentFinder->getIterator() : $currentFinder,
+                function (\SplFileInfo $current) use ($path, $isFile) {
+                    return $isFile
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: `$isFile` is constant, so we could extract ternary outside the lambda - then, when `false === $isFile`, we won't need to execute ternary multiple times

```php
$iterator = new \CallbackFilterIterator(
    $currentFinder instanceof \IteratorAggregate ? $currentFinder->getIterator() : $currentFinder,
    $isFile
        ? function (\SplFileInfo $current) use ($path) {
            return $current->getRealPath() === $path;
        }
        : function (\SplFileInfo $current) use ($path) {
            return 0 === strpos($current->getRealPath(), $path);
        }
);

File: src/Console/ConfigurationResolver.php [hunk 2]
```diff
@@ -406,38 +406,26 @@ private function resolveConfigPath()
...
+                        ? $current->getRealPath() === $path
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: as you are comparing `$this->path` to real path, you must ensure that `$this->path` is real as well

File: src/Console/ConfigurationResolver.php [hunk 3]
```diff
@@ -406,38 +406,26 @@ private function resolveConfigPath()
...
+                        : 0 === strpos($current->getRealPath(), $path);
```
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: what if
$current: `/foo/bar_baz`
$path: `/foo/bar`
?

File: src/Console/ConfigurationResolver.php (hunk 1 above)
Comment (gharlan) [inline src/Console/ConfigurationResolver.php]: I thought about it, too. But I guess there is no real performance impact, and imho the other solution is more readable.
But can change it, if you want.

Comment (gharlan) [inline src/Console/ConfigurationResolver.php]: should the string checks case insensitive?

**Comment by keradus** [comment]
Interesting approach,please take a look at my comments, then tomorrow I will play with your code ;)
Thank you for helping!

File: src/Console/ConfigurationResolver.php (hunk 3 above)
Comment (gharlan) [inline src/Console/ConfigurationResolver.php]: good point

File: src/Console/ConfigurationResolver.php (hunk 1 above)
Comment (keradus) [inline src/Console/ConfigurationResolver.php]: when your finder will found thousands of files then it's cool to not have so many conditions if it's possible ;)

when you have both side of comparison the realpath they will be same case.

**Comment by gharlan** [comment]
your mentioned problems should be fixed now.

**Comment by gharlan** [comment]
(but I only tested with you unit tests, didn't made any test by my own)

**Comment by keradus** [comment]
I enhanced tests for better testing intersection of iterators and your proposal works nice with it.

Thank you!

**Pull request closed without merging**